lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4828,7 +4827,8 @@ namespace {
       if (auto gpImportResult = importObjCGenericParams(decl, dc)) {
         auto genericParams = *gpImportResult;
         if (genericParams) {
-          result->setGenericParams(genericParams);
+          result->getASTContext().evaluator.cacheOutput(
```

Comment:
It seems that this comes up often enough that a setGenericParams() method is still useful. I would instead factor it so that GenericParamListRequest::cacheResult() calls setGenericParams().